lib/middleware/index.d.ts

```diff
@@ -124,20 +103,31 @@ declare namespace WatsonMiddleware {
       callback: (err: string | Error | null, watsonResponse?: Data) => void
     ) => void;
     sendToWatsonAsync: (
-      bot: Botkit.Bot<any, Botkit.Message>,
-      message: Botkit.Message,
+      bot: Botkit.BotWorker,
+      message: Botkit.BotkitMessage,
       contextDelta: ContextDelta
     ) => Bluebird<void>;
     interpretAsync: (
-      bot: Botkit.Bot<any, Botkit.Message>,
-      message: Botkit.Message
+      bot: Botkit.BotWorker,
+      message: Botkit.BotkitMessage
     ) => Bluebird<void>;
     readContextAsync: (user: string) => Bluebird<Context>;
     updateContextAsync: (user: string, context: Context) => Bluebird<Data>;
   }
 
-  export function createWatsonMiddleware(config: MiddlewareConfig): Middleware;
+  interface Payload {
+    workspace_id: string;
+    input: {
+      text: string;
+    };
+    context?: Context;
+  }
+
+  interface ContextDelta {
+    [index: string]: any;
+  }
+
+  export function factory(config: MiddlewareConfig): Middleware;
 }
 
-export default WatsonMiddleware.createWatsonMiddleware;
-export { WatsonMiddleware };
+export = WatsonMiddleware.factory;
```

Comment:
From reviewing this, you can just have `export { WatsonMiddleware };` at the end and then consumers can import it as follows:
`import { WatsonMiddleware } from 'botkit-middleware-watson';`
then consumers can reference the types as follows:
`let context: WatsonMiddleware.ContextDelta;`

Having said that, I don't often write TS definitions for Javascript source files, and exports seems to be a tricky area.

You can confirm this syntax works if you write a unit test in `.ts` which is probably a good idea, so you can prove the code will work in Javascript *and* Typescript.